src/interpolations.jl

```diff
@@ -683,75 +683,61 @@ function shape_value(ip::Lagrange{RefTriangle, 2}, ξ::Vec{2}, i::Int)
 end
 
 ######################################
-# Lagrange RefTriangle order 3, 4, 5 #
+# Lagrange RefTriangle order #
 ######################################
 # see https://getfem.readthedocs.io/en/latest/userdoc/appendixA.html
 
-const Lagrange2Tri345 = Union{
-    Lagrange{RefTriangle,3},
-    Lagrange{RefTriangle,4},
-    Lagrange{RefTriangle,5},
-}
-
-function getnbasefunctions(ip::Lagrange2Tri345)
-    order = getorder(ip)
-    return (order + 1) * (order + 2) ÷ 2
+function getnbasefunctions(::Lagrange{RefTriangle,N}) where N
+    return (N + 1) * (N + 2) ÷ 2
 end
 
 # Permutation to switch numbering to Ferrite ordering
-const permdof2DLagrange2Tri345 = Dict{Int,Vector{Int}}(
-    1 => [1, 2, 3],
-    2 => [3, 6, 1, 5, 4, 2],
-    3 => [4, 10, 1, 7, 9, 8, 5, 2, 3, 6],
-    4 => [5, 15, 1, 9, 12, 14, 13, 10, 6, 2, 3, 4, 7, 8, 11],
-    5 => [6, 21, 1, 11, 15, 18, 20, 19, 16, 12, 7, 2, 3, 4, 5, 8, 9, 10, 13, 14, 17],
-)
-
-function facedof_indices(ip::Lagrange2Tri345)
-    order = getorder(ip)
-    order == 1 && return ((1,2), (2,3), (3,1))
-    order == 2 && return ((1,2,4), (2,3,5), (3,1,6))
-    order == 3 && return ((1,2,4,5), (2,3,6,7), (3,1,8,9))
-    order == 4 && return ((1,2,4,5,6), (2,3,7,8,9), (3,1,10,11,12))
-    order == 5 && return ((1,2,4,5,6,7), (2,3,8,9,10,11), (3,1,12,13,14,15))
+function getPermLagrangeTri(order)
+    verts = SVector{3}(order + 1, (order + 1) * (order + 2) ÷ 2, 1)
+    edge1 = SVector{order-1}((sum(i:order+1) for i in order:-1:2))
+    edge2 = SVector{order-1}((sum(i:order+1)+1 for i in 3:(order+1)))
+    edge3 = SVector{order-1}((i for i in 2:order))
+    # TODO: fix this bottleneck, nested generators don't work so one has to use ...
+    interior = NTuple{(order - 2) * (order - 1) ÷ 2}((i for j in (order+1):-1:4 for i in sum(j:order+1)+2 : sum(j:order+1)+j-2))
+    return SVector((verts..., edge1..., edge2..., edge3..., interior...))
+end
```

Comment:
I think it is worth to investigate a code generator here in the future (e.g. to just generate the elements from the Ciarlet definition). So, no need to invest too much time here if it is not absurdely slow. :)

Reply:
Passing `Val(order)` might be a remedy